code/modules/status_system/statusEffects.dm

```diff
@@ -345,209 +345,111 @@
-				if(locate(/obj/item/implant/health) in M)
-					src.visible = 1
-				else
-					src.visible = 0
-				if (prob(stage - 2 - !!(M.traitHolder?.hasTrait("stablegenes"))) && (M.bioHolder && !M.bioHolder.HasEffect("revenant")))
-					boutput(M, "<span class='alert'>You mutate!</span>")
-					M.bioHolder.RandomEffect("either")
-				if(!ON_COOLDOWN(M, "radiation_stun_check", 1 SECOND) && prob(stage - 1) && M.bioHolder && !M.bioHolder.HasEffect("revenant"))
-					M.changeStatus("weakened", 3 SECONDS)
-					boutput(M, "<span class='alert'>You feel weak.</span>")
-					M.emote("collapse")
-
-			icon_state = "radiation[stage]"
-
-			return ..(timePassed)
-
-	simpledot/n_radiation
-		id = "n_radiation"
-		name = "Neutron Irradiated"
-		desc = ""
-		icon_state = "nradiation1"
-		unique = 1
-		visible = 0
-
-		tickSpacing = 2 SECONDS
-
-		damage_tox = 2
-		damage_brute = 2
-		damage_type = DAMAGE_STAB | DAMAGE_BURN
-
-		maxDuration = 4 MINUTES
-
-		var/howMuch = ""
-		var/stage = 1
-
-		getTooltip()
-			. = "You are [howMuch]irradiated by neutrons.<br>Taking [round(damage_tox, 0.1)] toxin damage every [tickSpacing/(1 SECOND)] sec and [round(damage_brute, 0.1)] brute damage every [tickSpacing/(1 SECOND)] sec.<br>Damage reduced by radiation resistance on gear."
-
-		preCheck(var/atom/A)
-			. = 1
-			if(issilicon(A) || isobserver(A) || isintangible(A))
-				. = 0
-		proc/get_stage(val)
-			. = 1
-			switch(val/(1 SECOND))
-				if(0 to 10)
-					. = 1
-				if(10 to 20)
-					. = 2
-				if(20 to 30)
-					. = 3
-				if(30 to 60)
-					. = 4
-				if(60 to INFINITY)
-					. = 5
-
-		modify_change(change)
-			. = change
-
-			if (owner && ismob(owner) && change > 0)
-				var/mob/M = owner
-				SEND_SIGNAL(M, COMSIG_MOB_GEIGER_TICK, get_stage(duration + change))
-				var/percent_protection = clamp(GET_ATOM_PROPERTY(M, PROP_MOB_RADPROT), 0, 100)
-				percent_protection = 1 - (percent_protection/100) //scale from 0 to 1
-				. *= percent_protection
-
-		proc/update_lights(add = 1)
-			owner.remove_simple_light("neutron_rad")
-			owner.remove_medium_light("neutron_rad")
-			if(add)
-				if(stage < 4)
-					owner.add_simple_light("neutron_rad", list(4, 62, 155, stage * 50))
-				else
-					owner.add_medium_light("neutron_rad", list(4, 62, 155, stage * 50))
-
-		onAdd(optional=null)
-			. = ..()
-			update_lights()
-
-		onRemove()
-			update_lights(0)
-			. = ..()
-
-		onChange(optional=null)
-			. = ..()
-			update_lights()
+			. = 0
+			switch(val/0.4) //0.4 Sv is radiation poisoning, 2 Sv is fatal in some cases, 4 Sv is fatal without treatment
+				if(0 to 0.1)
+					. = 0 //normal dose
+				if(0.1 to 1)
+					. = 1 //you might feel sick
+				if(1 to 3)
+					. = 2 //you're getting into dangerous teritory
+				if(3 to 5)
+					. = 3 //you're at a 50/50 of kicking it
+				if(5 to 7)
+					. = 4 //more like 70/30 now
+				if(7 to 10)
+					. = 5 //you will die without treatment
+				if(7 to INFINITY)
+					. = 6 //you will die.
```

Comment:
imo should pull the `/0.4` factor down into the switch branches for readability - dividing by `0.4` makes it annoying to mentally translate rad dose into rad stage